src/paths.rs

```diff
@@ -0,0 +1,123 @@
+use crate::db_state::SsTableId;
+use crate::db_state::SsTableId::{Compacted, Wal};
+use crate::error::SlateDBError;
+use object_store::path::Path;
+use ulid::Ulid;
+
+pub(crate) struct PathBuilder {
```

Comment:
nit: Would maybe something along the lines of `path resolver` be slightly more descriptive? Initially when I saw the name I thought it's a mutable datastructure, following the builder pattern.

```suggestion
pub(crate) struct PathResolver {
```

Reply:
Good call. I wasn't totally happy with "Builder."